src/Sentry.DiagnosticSource/Internal/DiagnosticSource/SentrySqlListener.cs

```diff
@@ -240,12 +225,6 @@ public void OnNext(KeyValuePair<string, object?> kvp)
                     TrySetConnectionStatistics(closeSpan, kvp.Value);
                     closeSpan.Finish(SpanStatus.Ok);
                     return;
-                case SqlMicrosoftWriteTransactionCommitAfter or SqlDataWriteTransactionCommitAfter
-                    when GetSpan(SentrySqlSpanType.Connection, kvp) is { } commitSpan:
-                    // If some query makes changes to the Database data, CloseAfterCommand event will not be invoked,
-                    // instead, TransactionCommitAfter is invoked.
-                    commitSpan.Finish(SpanStatus.Ok);
-                    break;
```

Comment:
This was the main culprit.  This event was finishing the connection span, so the case above it would log the warning inside its call to `GetSpan`